File: php/features.php
<?php
declare(strict_types=1);

// StrictlyBetter SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class StrictlyBetterFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new StrictlyBetterBaseFeature();
            case "test":
                return new StrictlyBetterTestFeature();
            default:
                return new StrictlyBetterBaseFeature();
        }
    }
}

File: py/features.py
# StrictlyBetter SDK feature factory

from feature.base_feature import StrictlyBetterBaseFeature
from feature.test_feature import StrictlyBetterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StrictlyBetterBaseFeature(),
        "test": lambda: StrictlyBetterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
